### PR TITLE
Fix type annotations

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Bugfix

### Purpose
- some type annotations has been wrong.

### CODE REVIEW

File: sphinx/application.py [hunk 1]
```diff
@@ -775,7 +775,7 @@ def add_index_to_domain(self, domain, index, override=False):
     def add_object_type(self, directivename, rolename, indextemplate='',
                         parse_node=None, ref_nodeclass=None, objname='',
                         doc_field_types=[], override=False):
-        # type: (unicode, unicode, unicode, Callable, nodes.Node, unicode, List, bool) -> None
+        # type: (unicode, unicode, unicode, Callable, Type[nodes.Node], unicode, List, bool) -> None
```
Comment (cocoatomo) [inline sphinx/application.py]: flake8 reports `E501 line too long (100 > 95 characters)` warning. `# NOQA` is needed.

File: sphinx/registry.py [hunk 2]
```diff
@@ -231,7 +231,7 @@ def add_index_to_domain(self, domain, index, override=False):
     def add_object_type(self, directivename, rolename, indextemplate='',
                         parse_node=None, ref_nodeclass=None, objname='',
                         doc_field_types=[], override=False):
-        # type: (unicode, unicode, unicode, Callable, nodes.Node, unicode, List, bool) -> None
+        # type: (unicode, unicode, unicode, Callable, Type[nodes.Node], unicode, List, bool) -> None
```
Comment (cocoatomo) [inline sphinx/registry.py]: flake8 reports `E501 line too long (100 > 95 characters)` warning. `# NOQA` is needed.

**Review by cocoatomo — CHANGES_REQUESTED** [review]

File: sphinx/domains/__init__.py [hunk 3]
```diff
@@ -212,7 +212,7 @@ def role(self, name):
         fullname = '%s:%s' % (self.name, name)
 
         def role_adapter(typ, rawtext, text, lineno, inliner, options={}, content=[]):
-            # type: (unicode, unicode, unicode, int, Inliner, Dict, List[unicode]) -> nodes.Node  # NOQA
+            # type: (unicode, unicode, unicode, int, Inliner, Optional[Dict], Optional[List[unicode]]) -> Tuple[List[nodes.Node], List[nodes.Node]]  # NOQA
```
Comment (cocoatomo) [inline sphinx/domains/__init__.py:215]: Sorry for my too-late-comment after merging, but added two `Optional` seems unnecessary.

Comment (tk0miya) [reply sphinx/domains/__init__.py:215]: Thanks, I'd confused.

**Pull request merged**